lib/client.js

```diff
@@ -158,6 +158,123 @@ class Client {
+   *
+   * @return {Promise<Object>} FHIR resource
+   */
+  create(params) {
+    return this.httpClient.post(params.resourceType, params.resource);
+  }
+
+  /**
+   * Delete a resource by FHIR id.
+   *
+   * @example
+   *
+   * // Using promises
+   * fhirClient.delete({ resourceType: 'Patient', id: 12345 })
+   *           .then((data) => { console.log(data); });
+   *
+   * // Using async
+   * let response = await fhirClient.delete({ resourceType: 'Patient', id: 12345 });
+   * console.log(response);
+   *
+   * @param {Object} params - The request parameters.
+   * @param {string} params.resourceType - The resource type (e.g. "Patient", "Observation").
+   * @param {string} params.id - The FHIR id for the resource.
+   *
+   * @return {Promise<Object>} FHIR resource
+   */
+  delete(params) {
+    return this.httpClient.delete(`${params.resourceType}/${params.id}`);
+  }
+
+  /**
+   * Update a resource by FHIR id.
+   *
+   * @example
+   *
+   * const updatedPatient = { resourceType: 'Patient',
+   *                          birthDate: '1948-04-14' }
+   * // Using promises
+   * fhirClient.update({ resourceType: 'Patient',
+   *                     id: 12345,
+   *                     resource: updatedPatient,
+   *                   })
+   *           .then((data) => { console.log(data); });
+   *
+   * // Using async
+   * let response = await fhirClient.update({ resourceType: 'Patient',
+   *                                          id: 12345,
+   *                                          resource: updatedPatient,
+   *                                        });
+   * console.log(response);
+   *
+   * @param {Object} params - The request parameters.
+   * @param {string} params.resourceType - The resource type (e.g. "Patient", "Observation").
+   * @param {string} params.id - The FHIR id for the resource.
+   * @param {string} params.resource - The resource to be updated.
+   *
+   * @return {Promise<Object>} FHIR resource
+   */
+  update(params) {
+    return this.httpClient.put(`${params.resourceType}/${params.id}`, params.resource);
+  }
+
+  /**
+   * Patch a resource by FHIR id.
+   *
+   * @example
+   *
+   * // JSON Patch document format from http://jsonpatch.com/
+   * const JSONpatch = [{ op: 'replace', path: '/gender', value: 'male' }];
+   *
+   * // Using promises
+   * fhirClient.patch({ resourceType: 'Patient', id: 12345, body: JSONpatch })
+   *           .then((data) => { console.log(data); });
+   *
+   * // Using async
+   * let response = await fhirClient.patch({ resourceType: 'Patient', id: 12345, body: JSONpatch });
+   * console.log(response);
+   *
+   * @param {Object} params - The request parameters.
+   * @param {string} params.resourceType - The resource type (e.g. "Patient", "Observation").
+   * @param {string} params.id - The FHIR id for the resource.
+   * @param {string} params.body - A JSON Patch document, formatted according to http://jsonpatch.com/.
```

Comment:
`axios.patch` consumes a string (and not a POJO?) for `body` ?

...if so, could the signature be changed to an Object, that then gets stringed?

(...and that Object could reference a `@typedef` that elucidates its structure?)